iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/subscription/SubscriptionCoordinator.java

```diff
@@ -80,6 +80,17 @@ public AtomicReference<SubscriptionInfo> tryLock() {
     return null;
   }
 
+  /**
+   * Lock the {@link SubscriptionInfo} coordinator.
+   *
+   * @return the {@link SubscriptionInfo} holder, which can be used to get the {@link
+   *     SubscriptionInfo}. Wait until lock is acquired
+   */
+  public AtomicReference<SubscriptionInfo> lock() {
+    coordinatorLock.lock();
+    return new AtomicReference<>(subscriptionInfo);
```

Comment:
Update subscriptionInfoHolder to prevent undefined behavior in procedure after unlocking